seqr/views/utils/pedigree_info_utils_2_3_tests.py

```diff
@@ -73,19 +83,119 @@ def test_parse_pedigree_table(self):
+            'Error while parsing file: {}. Expected vs. actual header columns: | Sample ID| Family ID| Alias|-Alias|-Paternal Sample ID| Maternal Sample ID| Gender| Affected Status'.format(
+                FILENAME)])
+        self.assertListEqual(warnings, [])
+        self.assertListEqual(records, [])
+
+        records, errors, warnings = parse_pedigree_table([
+            header_1, header_2, ['', 'Position', '', '', 'Collaborator Sample ID', '', '', '', '', 'ul', 'ng/ul', '',
+                                 '', 'indicate study/protocol number']], FILENAME)
+        self.assertListEqual(errors, [
+            'Error while parsing file: {}. Expected vs. actual header columns: |-Collaborator Participant ID| Collaborator Sample ID|+'.format(
+                FILENAME)])
+        self.assertListEqual(warnings, [])
+        self.assertListEqual(records, [])
+
+        original_data = [
+            header_1, header_2, header_3,
+            ['SK-3QVD', 'A02', 'SM-IRW6C', 'PED073', 'SCO_PED073B_GA0339', 'SCO_PED073B_GA0339_1', '', '', 'male',
+             'unaffected', '20', '94.8', 'probably dad', '', '1234'],
+            ['SK-3QVD', 'A03', 'SM-IRW69', 'PED073', 'SCO_PED073C_GA0340', 'SCO_PED073C_GA0340_1',
+             'SCO_PED073B_GA0339_1', 'SCO_PED073A_GA0338_1', 'female', 'affected', '20', '98', '', 'Perinatal death', ''
+             ]]
+
+        records, errors, warnings = parse_pedigree_table(
+            original_data, FILENAME, user=User.objects.get(id=10), project=Project.objects.get(id=1))
+        self.assertListEqual(records, [
+            {'affected': 'N', 'maternalId': '', 'notes': 'probably dad', 'individualId': 'SCO_PED073B_GA0339_1',
+             'sex': 'M', 'familyId': 'PED073', 'paternalId': '', 'codedPhenotype': ''},
+            {'affected': 'A', 'maternalId': 'SCO_PED073A_GA0338_1', 'notes': '', 'individualId': 'SCO_PED073C_GA0340_1',
+             'sex': 'F', 'familyId': 'PED073', 'paternalId': 'SCO_PED073B_GA0339_1', 'codedPhenotype': 'Perinatal death'
+             }])
+        self.assertListEqual(
+            warnings,
+            ["SCO_PED073A_GA0338_1 is the mother of SCO_PED073C_GA0340_1 but doesn't have a separate record in the table"])
+        self.assertListEqual(errors, [])
+
+        mock_email.assert_called_with(
+            subject='SK-3QVD Merged Sample Pedigree File',
+            body=mock.ANY,
+            to=['recipient@test.com'],
+            attachments=[
+                ('SK-3QVD.xlsx', mock.ANY,
+                 "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"),
+                ('test.xlsx', mock.ANY,
+                 "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"),
+            ])
+        self.assertEqual(
+            mock_email.call_args.kwargs['body'],
+            """User test_user@test.com just uploaded pedigree info to 1kg project n\xe5me with uni\xe7\xf8de.This email has 2 attached files:
+    
```

Comment:
Codacy found an issue: [Trailing whitespace](https://app.codacy.com/manual/hanars/seqr/pullRequest?prid=5795824)

Reply:
whitespace is part of the test